File: HRMS.Desktop/lib/models/requests/position_insert_update.dart
import 'package:json_annotation/json_annotation.dart';

part 'position_insert_update.g.dart';

@JsonSerializable()
class PositionInsertUpdate {
  String name;
  int? departmentId;
  int? payGradeId;
  int? requiredEducationId;
  bool isWorkExperienceRequired;

  PositionInsertUpdate(
    this.name,
    this.departmentId,
    this.payGradeId,
    this.requiredEducationId,
    this.isWorkExperienceRequired,
  );

  factory PositionInsertUpdate.fromJson(Map<String, dynamic> json) =>
      _$PositionInsertUpdateFromJson(json);

  Map<String, dynamic> toJson() => _$PositionInsertUpdateToJson(this);
}

File: HRMS/Models/PositionInsertUpdate.cs
using System.ComponentModel.DataAnnotations;

namespace HRMS.Models;

public class PositionInsertUpdate
{
    [Required]
    public string Name { get; set; } = string.Empty;

    [Required]
    public int? DepartmentId { get; set; }

    [Required]
    public int? PayGradeId { get; set; }

    [Required]
    public int? RequiredEducationId { get; set; }

    public bool IsWorkExperienceRequired { get; set; }
}